pkg/events/events.go

```diff
@@ -5815,6 +5815,11 @@ var Definitions = eventDefinitions{
 					{MapName: "prog_array", MapIndexes: []uint32{tailKernelWrite}, ProgName: "trace_ret_kernel_write_tail"},
 					{MapName: "prog_array", MapIndexes: []uint32{tailSendBin}, ProgName: "send_bin"},
 				},
+				Events: []eventDependency{
+					{VfsWrite},
+					{VfsWritev},
```

Comment:
1. There are environments (GKE if I remember correctly) where this event will fail to attach since the vfs_writev symbol is not exposed through kallsyms - won't this make capture unavailable on such environments?
2. vfs_write and vfs_writev can be very verbose, and sending them through the events perf buffer may cause lost events. I'm not sure this is the best way to go (maybe we should consider sending the path when we capture in the bpf code instead of adding these dependencies)

Reply:
> 1. There are environments (GKE if I remember correctly) where this event will fail to attach since the vfs_writev symbol is not exposed through kallsyms - won't this make capture unavailable on such environments?

You are right that this is a problem, but it is somewhat impossible to really use symbols that are available for all the kernels (`vfs_writev` is a very basic symbol, so I think we should be able to expect it to be available most of the time). The only solution is to tell somehow that it is okay for the dependency to fail (maybe add require flag?)

> 2. vfs_write and vfs_writev can be very verbose, and sending them through the events perf buffer may cause lost events. I'm not sure this is the best way to go (maybe we should consider sending the path when we capture in the bpf code instead of adding these dependencies)

I agree with you here, and I think this solution is the best. But are we okay with letting a bug with one of our features to stay?